src/internal/connector/onedrive/drive.go

```diff
@@ -172,22 +183,41 @@ func collectItems(
 	pager itemPager,
 	driveID, driveName string,
 	collector itemCollector,
-) (string, map[string]string, map[string]struct{}, error) {
+	prevDelta string,
+) (DeltaUpdate, map[string]string, map[string]struct{}, error) {
 	var (
 		newDeltaURL = ""
 		// TODO(ashmrtn): Eventually this should probably be a parameter so we can
 		// take in previous paths.
-		oldPaths = map[string]string{}
-		newPaths = map[string]string{}
-		excluded = map[string]struct{}{}
+		oldPaths         = map[string]string{}
+		newPaths         = map[string]string{}
+		excluded         = map[string]struct{}{}
+		invalidPrevDelta = false
+		triedPrevDelta   = false
 	)
 
 	maps.Copy(newPaths, oldPaths)
 
+	if len(prevDelta) != 0 {
+		pager.SetNext(prevDelta)
+	}
+
 	for {
 		page, err := pager.GetPage(ctx)
+
+		if !triedPrevDelta && graph.IsErrInvalidDelta(err) {
+			logger.Ctx(ctx).Infow("Invalid previous delta link", "link", prevDelta)
+
+			triedPrevDelta = true // TODO(meain): Do we need this check?
```

Comment:
> Do we need this check?

probably not

Reply:
Yeah, I guess this specific error is only if the delta is invalid. I am being paranoid.